docs/proposed-api.md

```diff
@@ -288,3 +288,154 @@ spec:
             - not: 192.168.0.0/16
             - not: 3.2.1.0/30
 ```
+
+## Simple firewall implementation
+
+VPC Firewall implies zero trust if firewall pattern is specified.
```

Comment:
Do I understand this right?:

- No `firewall` policy specified: all traffic allowed
- `allow` policy specified: all traffic denied, save for allowed patterns
- `deny`-only policy specified: all traffic denied
- `allow` + `deny` policies: traffic not matching allowed patterns is denied; traffic matching allowed patterns is allowed, except if it's explicitly denied by the `deny` policy, which always takes precedence

Is this correct?

Reply:
Yes. If there is `deny` only policy specified, `allow` should be added for any other traffic.
Also, order is defined by order in the API message.

Reply:
> Yes. If there is `deny` only policy specified, `allow` should be added for any other traffic.

Sounds good to me, thanks!

> Also, order is defined by order in the API message.

This is more concerning to me. It sounds super-easy to enable by mistake the traffic you specifically wanted to deny, because a user didn't understand the significance of the order, or they got confused and mentally swapped the order, or did a copy-paste error and put the deny policy at the wrong place. Are we sure we need to nest multiple levels of allow and deny policies? Wouldn't a single level of `allow` + a preceding level of `deny` be safer (and easier) to use?